backend/codesirius/problems/signals.py

```
@@ -0,0 +1,19 @@
+# problems/signals.py
+from django.db.models.signals import m2m_changed
+from django.dispatch import receiver
+from problems.models.problem import Problem
+from problems.models.execution_constraint import ExecutionConstraint
+
+
+@receiver(m2m_changed, sender=Problem.languages.through)
+def remove_execution_constraints_on_language_removal(
```

Ruff (0.8.2)

Missing return type annotation for public function `remove_execution_constraints_on_language_removal`

Add return type annotation: `None`

(ANN201)